### ISSUE TITLE
[RFC] Adapter folder structure

### ISSUE THREAD

**Issue by jamwaffles** [repositive/event-store-rs#57]
In #53, the store/cache/emitter adapter folder structure was changed somewhat. Instead of grouping by backend driver (`adapters/amqp.rs`, `adapters/pg.rs`, etc), they are grouped by function (`adapters/cache/pg.rs`, `adapters/emitter/amqp.rs`). It turns out this was easier to work with and, in my opinion, makes more sense as adapters are grouped by function, and each adapter's `mod.rs` can contain just the trait for that adapter.

Therefore, I propose changing the folder structure of the adapters from:

```
src/adapters
├── mod.rs
├── amqp
│   ├── emitter.rs
│   └── mod.rs
├── pg
│   ├── cache.rs
│   ├── mod.rs
│   └── store.rs
└── stub
    ├── emitter.rs
    └── mod.rs
```

To:

```
src/adapters
├── mod.rs
├── emitter
│   ├── amqp.rs
│   ├── stub.rs
│   └── mod.rs
├── cache
│   ├── pg.rs
│   ├── mod.rs
└── store
    ├── pg.rs
    └── mod.rs
```

In addition, the traits contained in `src/adapters/mod.rs` should be moved into their new associated `src/adapters/<type>/mod.rs` files.

**Comment by Istar-Eldritch** [comment]
:+1: Go for it.

**Comment by jamwaffles** [comment]
Closed by #59
